src/utils/common_utils.hpp

```diff
@@ -51,6 +54,48 @@ bool make_path(const std::string& path);
 /// Check if directory with given path exist
 bool is_dir_exist(const std::string& path);
 
+/// Generate random std::string of length len based on a
+/// uniform distribution
+std::string generate_random_string(int len);
+
+/// Singleton class for random strings that are appended to the
+/// Eigen matrices names that are used in the solutions of
+/// nmodl::visitor::SympySolverVisitor and need to be the same to
+/// be printed by the nmodl::codegen::CodegenCVisitor
+template <unsigned int SIZE = 4>
+class SingletonRandomString {
+  public:
+    SingletonRandomString(SingletonRandomString const&) = delete;
+    SingletonRandomString& operator=(SingletonRandomString const&) = delete;
+
+    static std::shared_ptr<SingletonRandomString> instance() {
+        static std::shared_ptr<SingletonRandomString> s{new SingletonRandomString};
+        return s;
+    }
+
+    bool random_string_exists(const std::string& var_name) {
```

Comment:
```suggestion
    bool random_string_exists(const std::string& var_name) const {
```

Reply:
Also, to avoid having to later do doxygen campaigns, let's as a rule briefly document all public interfaces at least.

Reply:
You are right :+1: I will add documentation in all public interfaces from now on